admin/controllers/administrator.js

```diff
@@ -328,11 +352,39 @@ const saveCheckWindows = async (req, res, next) => {
   return res.redirect('/administrator/check-windows')
 }
 
+const removeCheckWindow = async (req, res, next) => {
+  let checkWindow
+
+  if (!req.params.checkWindowId) {
+    return res.redirect('/administrator/check-windows')
+  }
+
+  try {
+    checkWindow = await fetchCheckWindow(req.params.checkWindowId)
+  } catch (err) {
+    return next(err)
+  }
+
+  if (checkWindow) {
+    if (Date.parse(checkWindow.checkStartDate) < Date.now()) {
+      return res.redirect('/administrator/check-windows/status/validationError')
```

Comment:
Usually you would write the flash message to session storage here - which happens automatically.   Then you do not need to to redirect to different routes.

Reply:
Removing is not rendering so I have two options:
1) Build the whole default page again in this POST request (not a great plan)
2) Redirect (not great but less bad)
Does it make sense?